### PR TITLE
[master] Close #1996 : use removeCache

### LINKED ISSUE #1996: destroyCache should use the "regular" removeCache method
in [EhcacheManager#683](https://github.com/ehcache/ehcache3/blob/e496850c8bf8191a30d7a783c80944c56d2cdc74/core/src/main/java/org/ehcache/core/EhcacheManager.java#L683) , ```destroyCache``` is calling ```removeAndCloseWithoutNotice``` that unfortunately does not trigger ```cacheRemoved```.

It can be problematic, since the user can destroy a cache programmatically without the monitoring capabilities knowing that a cache was indeed removed

### PR DESCRIPTION
**Pull request by anthonydahanne**

### CODE REVIEW

File: core/src/main/java/org/ehcache/core/EhcacheManager.java [hunk 1]
```diff
@@ -680,7 +680,7 @@ public void destroyCache(final String alias) throws CachePersistenceException {
     }
 
     try {
-      removeAndCloseWithoutNotice(alias);
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: This method is no longer needed, remove it

File: core/src/main/java/org/ehcache/core/EhcacheManager.java [hunk 2]
```diff
@@ -680,7 +680,7 @@ public void destroyCache(final String alias) throws CachePersistenceException {
...
+      removeCache(alias);
```
Comment (ljacomet) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: You should be using `removeCache(alias, true)`

File: core/src/main/java/org/ehcache/core/EhcacheManager.java (hunk 1 above)
Comment (anthonydahanne) [inline core/src/main/java/org/ehcache/core/EhcacheManager.java]: done

**Comment by ljacomet** [comment]
test this please

**Comment by ljacomet** [comment]
Time out in one build - merging anyway

**Pull request merged**